lib/common.py

```diff
@@ -677,6 +697,55 @@ def start_background_process(self, cmd):
     def pid_exists(self, pid):
         return self.ssh_with_result(['test', '-d', '/proc/%s' % pid]).returncode == 0
 
+    def execute_script(self, script_contents, simple_output=True):
+        with tempfile.NamedTemporaryFile('w') as f:
+            f.write(script_contents)
+            f.flush()
+            self.scp(f.name, f.name)
+            try:
+                res = self.ssh(['sh', f.name], simple_output=simple_output)
+                return res
+            finally:
+                self.ssh(['rm', '-f', f.name])
+
+    def distro(self):
+        """ Returns the distro name as detected by the guest tools.
+            If the distro name was not detected, the result will be an empty string
+        """
+        script = "eval $(xe-linux-distribution)\n"
+        script += "echo $os_distro\n"
+        return self.execute_script(script)
+
+    def tools_version_dict(self):
+        """ Returns the guest tools version as detected by the guest tools,
+            as a {major:, minor:, micro:, build:} dict. Values are strings.
+        """
+        return parse_xe_dict(self.param_get('PV-drivers-version'))
+
+    def tools_version(self):
+        """ Returns the tools version in the form major.minor.micro-build """
+        version_dict = self.tools_version_dict()
+        return "{major}.{minor}.{micro}-{build}".format(**version_dict)
+
+    def file_exists(self, filepath):
+        '''Test that the file at filepath exists.'''
```

Comment:
Why simple quote here instead of double as in other pydocs?

Reply:
We probably should decide on one and use always the same, even for strings in the code.

Reply:
Because I copy-pasted that function from Bobbye's PR :D
I'll fix it

Reply:
> We probably should decide on one and use always the same, even for strings in the code.

I usually use double quotes for multi word strings and single quotes for single-word strings, but that's debatable.